modules/material-management/src/multi_keyring.ts

```diff
@@ -113,19 +113,29 @@ function buildPrivateOnDecrypt<S extends SupportedAlgorithmSuites> () {
     const children = this.children.slice()
     if (this.generator) children.unshift(this.generator)
 
+    let keyringError = false
+
     for (const keyring of children) {
     /* Check for early return (Postcondition): Do not attempt to decrypt once I have a valid key. */
       if (material.hasValidKey()) return material
 
       try {
         await keyring.onDecrypt(material, encryptedDataKeys)
       } catch (e) {
-      // there should be some debug here?  or wrap?
       // Failures onDecrypt should not short-circuit the process
       // If the caller does not have access they may have access
       // through another Keyring.
+        keyringError = true
       }
     }
+
+    /* Postcondition: Need either a valid data key or no errors.
+     * If I have a key, decrypt errors should be ignored.
+     * However, if I was unable to decrypt a key AND I have errors,
+     * these errors should bubble up.
+     */
+    needs(material.hasValidKey() || !keyringError, 'Keyring errors and no keyring was successfully able to decrypt the encrypted data key.')
```

Comment:
naive JS question: How does JS handle the bubbling up of errors? If this needs condition is not met due to an error caught on L122, how does the error (and possibly others) propagate up, and how is that error handled?

Reply:
The errors from L122 are lost and not wrapped.  That is not a bad idea to add, but it will require some other re-work